### ISSUE TITLE
How to use beet in node.js modules

### ISSUE THREAD

**Issue by ricardocr987** [metaplex-foundation/beet#45]
When I use this library I get this error, executing the transpiled code of my project:

<img width="923" alt="Screenshot 2022-09-19 at 13 00 05" src="https://user-images.githubusercontent.com/32191898/191005509-d44d038e-de8d-480a-be1a-0485c184d988.png">

It seems like it treats it as if it were transpiled code with the CommonJS format and that's because I think the esm resolution fails.

If I modify the package.json of beet's (and beet-solana) node_modules including: "type": "module", I stop getting the error, because that's how it seems to specify that the module uses the esm format.

This is the tsconfig.json of the project I'm working:
```
{
  "compilerOptions": {
    "target": "ES2021",
    "lib": [
      "ESNext",
    ],
    "module": "esnext",
    "esModuleInterop": true,
    "forceConsistentCasingInFileNames": true,
    "strict": true,
    "skipLibCheck": true,
    "declaration": true,
    "moduleResolution": "node",
    "resolveJsonModule": true,
    "outDir": "dist",
    "declarationMap": true,
    "allowJs": false,
  },
  "exclude": [
    "node_modules",
    "dist",
    "scripts",
    "**/*.spec.ts",
    "**/*.test.ts",
    "**/__tests__",
    "**/__mocks__"
  ]
}
```

And this is the package.json:

```
{
  "name": "",
  "version": "1.0.0",
  "description": "",
  "main": "dist/run.js",
  "module": "dist/run.js",
  "types": "dist/types.d.ts",
  "type": "module",
  "scripts": {
    "start": "npm i && npm run build && node ./dist/run.js",
    "build": "tsc -p ./tsconfig.json",
    "clean:ts": "rm -rf ./dist",
    "clean:all": "rm -rf ./node_modules && rm -rf ./dist && rm -rf package-lock.json",
    "test": "echo \"Error: no test specified\" && exit 1",
  },
  "author": "",
  "license": "ISC",
  "dependencies": {
    "@metaplex-foundation/beet": "0.6.1",
    "@metaplex-foundation/beet-solana": "0.3.1",
    "@solana/web3.js": "1.61.1",
    "bs58": "5.0.0",
    "graphql": "16.6.0",
    "graphql-tools": "8.3.6",
    "@solana/spl-token": "0.3.5"
  },
  "devDependencies": {
    "@types/luxon": "^3.0.1",
    "@types/node": "^18.7.18",
    "typescript": "^4.8.3"
  }
}
```

**Comment by thlorenz** [comment]
This should get fixed shortly via removing explicit ESM support but relying solely on CJS.
@lorisleiva is fixing this ASAP.

**Comment by lorisleiva** [comment]
Perfect timing haha #47

**Comment by thlorenz** [comment]
Can you please see if this works with beet `v0.7.1` and let us know? Thanks.

**Comment by ricardocr987** [comment]
Yep, thank you so much guys, but now I have the same problem with beet-solana, I noticed that you have removed the repository, in that case are you going to include the PublicKey type in beet or what are you planning to do regarding this issue?

**Comment by thlorenz** [comment]
Sorry you have that problem, but the repo is not removed (it's part of this monorepo).

Maybe you need to just delete your yarn.lock + remove node_modules and re-install everything since it could be you still got an older beet lying around.
beet-solana auto updates to latest available beet when installed [see here](https://github.com/metaplex-foundation/beet/blob/master/beet-solana/package.json#L32)

**Comment by ricardocr987** [comment]
it's true 🤦🏻‍♂️🤦🏻‍♂️, looked for the package in npm that points to a repository that doesn’t exist: https://www.npmjs.com/package/@metaplex-foundation/beet-solana

I still have the same problem anyway. Removed node_modules and package-lock.json

![Screenshot 2022-09-20 at 23 52 11](https://user-images.githubusercontent.com/32191898/191371647-1e32c27a-7748-4fa9-840c-874f0e261d5f.png)

The package versions are also in the latest version:
    "@metaplex-foundation/beet": "0.7.1",
    "@metaplex-foundation/beet-solana": "0.3.1",

**Comment by thlorenz** [comment]
Oh sorry you're right. beet-solana also had esm support, got removed, but I hadn't published a new version.

That is fixed now, please upgrade to `v0.4.0`.

**Comment by ricardocr987** [comment]
Thank you so much, is working now :)
